types/dom-navigation/index.d.ts

```diff
@@ -0,0 +1,182 @@
+        type: K,
+        listener: (this: Navigation, ev: NavigationEventMap[K]) => any,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    addEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    removeEventListener<K extends keyof NavigationEventMap>(
+        type: K,
+        listener: (this: Navigation, ev: NavigationEventMap[K]) => any,
+        options?: boolean | EventListenerOptions,
+    ): void;
+    removeEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | EventListenerOptions,
+    ): void;
+}
+
+declare class NavigationTransition {
+    readonly navigationType: NavigationTypeString;
+    readonly from: NavigationHistoryEntry;
+    readonly finished: Promise<void>;
+}
+
+interface NavigationHistoryEntryEventMap {
+    dispose: Event;
+}
+
+declare class NavigationHistoryEntry extends EventTarget {
+    readonly key: string;
+    readonly id: string;
+    readonly url: string | null;
+    readonly index: number;
+    readonly sameDocument: boolean;
+
+    getState(): unknown;
+
+    ondispose: ((this: NavigationHistoryEntry, ev: Event) => any) | null;
+
+    addEventListener<K extends keyof NavigationHistoryEntryEventMap>(
+        type: K,
+        listener: (this: NavigationHistoryEntry, ev: NavigationHistoryEntryEventMap[K]) => any,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    addEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    removeEventListener<K extends keyof NavigationHistoryEntryEventMap>(
+        type: K,
+        listener: (this: NavigationHistoryEntry, ev: NavigationHistoryEntryEventMap[K]) => any,
+        options?: boolean | EventListenerOptions,
+    ): void;
+    removeEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | EventListenerOptions,
+    ): void;
+}
+
+type NavigationTypeString = 'reload' | 'push' | 'replace' | 'traverse';
+
+interface NavigationUpdateCurrentEntryOptions {
+    state: unknown;
+}
+
+interface NavigationOptions {
+    info?: unknown;
+}
+
+interface NavigationNavigateOptions extends NavigationOptions {
+    state?: unknown;
+    history?: 'auto' | 'push' | 'replace';
+}
+
+interface NavigationReloadOptions extends NavigationOptions {
+    state?: unknown;
+}
+
+declare class NavigationCurrentEntryChangeEvent extends Event {
+    constructor(type: string, eventInit?: NavigationCurrentEntryChangeEventInit);
+
+    readonly navigationType: NavigationTypeString | null;
+    readonly from: NavigationHistoryEntry;
+}
+
+interface NavigationCurrentEntryChangeEventInit extends EventInit {
+    navigationType?: NavigationTypeString | null;
+    from: NavigationHistoryEntry;
```

Comment:
This is [`destination`](https://wicg.github.io/navigation-api/#dom-navigationcurrententrychangeeventinit-destination) in the specification, but from context this is probably a specification bug as `from` is what we have in Chromium.

Reply:
I'll mention this to the spec maintainers.